cmd/argo/commands/common.go

```diff
@@ -70,7 +70,8 @@ func initKubeClient() *kubernetes.Clientset {
 	return clientset
 }
 
-func initWorkflowClient(ns ...string) *wfclient.WorkflowClient {
+// InitWorkflowClient creates a new client for the Kubernetes Workflow CRD.
+func InitWorkflowClient(ns ...string) *wfclient.WorkflowClient {
```

Comment:
I'm a bit weary exposing these commands publicly for sole the purpose of making them available in our e2e tests, but as of now I can't offer a better solution so I think this is fine for now.

Reply:
What about creating a public interface and changing the function signature to return that? Then the tests could bring their own implementation with stubbed out network calls and stuff.

Reply:
Let's address this in a separate commit. Stubbing out network calls, especially the ones that interact with the Kubernetes api server would be super useful; can increase test coverage as well as make tests more reliable.